src/lib/sputter/net/port.cpp

```diff
@@ -0,0 +1,108 @@
+#include "port.h"
+
+#include <cstring>
+#include <iostream>
+#include <netinet/in.h>
+#include <arpa/inet.h>
+#include <sys/socket.h>
+#include <unistd.h>
+
+using namespace sputter::net;
+
+UDPPort::UDPPort()
+    : socketHandle(-1) {
+}
+
+UDPPort::~UDPPort() {
+  close();
+}
+
+bool UDPPort::open(uint16_t port) {
```

Comment:
I had a lot of help from an AI for this. Critique is welcome

Reply:
lol nice